cockatrice/src/dlg_connect.cpp

```diff
@@ -156,26 +167,104 @@ void DlgConnect::actSaveConfig()
                                               passwordEdit->text(), savePasswordCheckBox->isChecked());
     }
 
-    rebuildComboBoxList();
+    preRebuildComboBoxList();
 }
 
-void DlgConnect::rebuildComboBoxList()
+void DlgConnect::downloadPublicServers()
```

Comment:
I'm not sure this belongs in here - it seems appropriate to be inside of an object whose only responsibility is to download the server data and fire a signal on success/fail completion

Reply:
This can probably be refactored to an external file, and I can look into that if you think it's worth it